Tests/JobsTests/JobsTests.swift

```diff
@@ -414,4 +414,45 @@ final class JobsTests: XCTestCase {
             XCTAssertEqual(date.advanced(by: offset).timeIntervalSinceReferenceDate, newDate.timeIntervalSinceReferenceDate, accuracy: 0.001)
         }
     }
+
+    func testPausedCancelledAndThenResume() async throws {
```

Comment:
Can we split this into two tests one for pause and resume and one for cancelling